e2e/bug-12972b/autoloader.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+require_once __DIR__ . '/vendor/autoload.php';
+
+spl_autoload_register(function($class) {
+	if ($class === \other12972\MyClass::class) {
+		throw new LogicException('this should not happen');
```

Comment:
in our real world project this autoloader is scanning some paths in the filesystem (similar to how a include-path worked in the past / or how the %PATH% variable works in a shell).

since it scans the filesystem it finds some files and includes them. this in turn leads to the problem described in https://github.com/phpstan/phpstan/issues/12972